css-nav-1/Overview.bs

```diff
@@ -1236,38 +1236,100 @@ follow the following steps:
+        3. Let <var>intersectRect</var> be an <dfn>intersecting area</dfn> between the search origin and a candidate.
+        4. Select the <var>P1</var> and <var>P2</var> that make the smallest <var>distance</var> between those two points by 
+            <a>calculating the distance</a>.        
+        5. Return the item of the <var>candidates</var> set that has the smallest <var>distance</var>.
             If several have the same distance,
             return the first one in document order,
             unless its [=boundary box=] overlaps with the [=boundary box=] of an other item at the same distance,
             and that item is higher in the CSS [=painting order=].
             In that case, return that item instead,
-            unless it too is overlapped with another higher item at the same distance, recursively.
+            unless it too is overlapped with another higher item at the same distance, recursively.    
+</div>
+
+<div algorithm="to calculate the distance">
+To <dfn lt="calculating the distance | to calculate the distance">calculate the <var>distance</var></dfn>, the function is defined as below:
+
+<pre class="prod">
+    <var>distance</var> = <var>euclidian</var> + <var>displacement</var> - <var>alignment</var> - <var>sqrt(Overlap)</var>
+</pre>
+
+The meaning of each factor in the function is as follows:
+
+<dl dfn-type="value" dfn-for="spatial-navigation-distance-function">
+    <dt><var>euclidian</var>
+    <dd>The euclidian distance between <var>P1</var> and <var>P2</var>
+
+    <dt><var>displacement</var>
+    <dd>The degree of displacement in <var>dir</var> between <var>searchOrigin</var> and <var>candidate</var>
+    <dd>Defined as <pre class="prod">(absolute distance on the axis orthogonal to <var>dir</var> + <var>orthogonalBias</var>) * <var>orthogonalWeight</var></pre>
+
+        * <var>orthogonalBias</var>
+            * If the <var>dir</var> is {{SpatialNavigationDirection/left}} or {{SpatialNavigationDirection/right}}, 
+            height of <var>searchOrigin</var> / 2
+            * Else if the <var>dir</var> is {{SpatialNavigationDirection/up}} or {{SpatialNavigationDirection/down}}, 
+            width of <var>searchOrigin</var> / 2
+
+        * <var>orthogonalWeight</var>:
+            * If the <var>dir</var> is {{SpatialNavigationDirection/left}} or {{SpatialNavigationDirection/right}}, 30
+            * Else if the <var>dir</var> is {{SpatialNavigationDirection/up}} or {{SpatialNavigationDirection/down}}, 2
+
+    <dt><var>alignment</var>
+    <dd>The degree of alignment in <var>dir</var> between <var>searchOrigin</var> and <var>candidate</var>
+    <dd>Defined as <pre class="prod"><var>alignBias</var> * <var>alignWeight</var></pre>
 
+        * <var>alignBias</var>
 
+            * If the <var>dir</var> is {{SpatialNavigationDirection/left}} or {{SpatialNavigationDirection/right}}, 
+            height of <var>intersectRect</var> / height of <var>candidate</var>
```

Comment:
In #3388 you wrote "length of the edge of the search origin". Here you wrote _candidate_'s edge" instead... Mistake?

Reply:
The edge of search origin on the comment (#3388) has been replaced with the current expression on the spec which describes an edge of candidate. If you have another opinion, please put it here so that we could consider. With minor updates (figures of some weights), we are going to apply the updated expression into Blink.

BTW, the polyfill would operate it with the edge of search origin that seems not to match with the current spec. @jihyerish please check and update on it.
https://github.com/WICG/spatial-navigation/blob/master/polyfill/spatial-navigation-polyfill.js#L1059

Reply:
I created [crbug.com/958845](https://crbug.com/958845) for the Blink work. 

See this example:

```
   ____________________ top edge of container or other focusable. 20 units wide.
   YYYYY XXXXXXXXXXXX
```
Here we want X to win so we do 12/20 .

`12 / 20 > 5 / 20` => we give X a **larger** `alignment` because we want to decrease the distance to X more than we decrease the distance to Y. (`alignment` is a negative term in the formula).

Reply:
Yes, you're right. My mistake...
Thanks for pointing this out! I'll fix this.

Reply:
This has been fixed by https://github.com/w3c/csswg-drafts/commit/bfe77ef53581fafe9a2a97696ef0413f8910e3d4.